n2y/main.py

```diff
@@ -61,67 +65,83 @@ def name_column_valid(raw_rows, name_column):
-            if pandoc_output:
-                markdown = pandoc.write(pandoc_output, format='gfm+tex_math_dollars') \
-                    .replace('\r\n', '\n')  # Deal with Windows line endings
-
-                # create target path if it doesn't exist
-                # if not os.path.exists(options.target):
-                os.makedirs(options.target, exist_ok=True)
-
-                # sanitize file name just a bit
-                # maybe use python-slugify in the future?
-                with open(os.path.join(options.target, f"{filename}.md"), 'w') as f:
-                    f.write('---\n')
-                    f.write(yaml.dump(meta))
-                    f.write('---\n\n')
-                    f.write(markdown)
+            filename = re.sub(r"[\s/,\\]", '_', page_name.lower())
+            if filename not in file_names:
+                file_names.append(filename)
+
+                pandoc_output = converter.load_block(client, row['id']).to_pandoc()
+                # do not create markdown pages if there is no page in Notion
+                if pandoc_output:
+                    logger.warning(f'Processing Page "{meta[options.name_column]}"')
+                    markdown = pandoc.write(pandoc_output, format='gfm+tex_math_dollars') \
+                        .replace('\r\n', '\n')  # Deal with Windows line endings
+
+                    # create target path if it doesn't exist
+                    # if not os.path.exists(options.target):
+                    os.makedirs(options.target, exist_ok=True)
+
+                    # sanitize file name just a bit
+                    # maybe use python-slugify in the future?
+                    with open(os.path.join(options.target, f"{filename}.md"), 'w') as f:
+                        f.write('---\n')
+                        f.write(yaml.dump(meta))
+                        f.write('---\n\n')
+                        f.write(markdown)
+                else:
+                    logger.debug(f"Skipping Empty Page: {page_name}")
+                    skips['Empty'] += 1
             else:
-                print(f"WARNING: skipping empty page: {page_name}", file=sys.stderr)
+                logger.debug(
+                    "%s %s",
+                    f'Skipping Duplicate Page: "{page_name}",',
+                    'Name Has Already Been Used. Please Rename')
+                skips['Duplicate'] += 1
         else:
-            print("WARNING: skipping page with no name", file=sys.stderr)
+            logger.debug("Skipping Page With No Name")
+            skips['Unnamed'] += 1
+    msg = ""
+    types_skipped = 0
+    prefixes = ("", " & ", ", & ")
+    for key in skips.keys():
+        count = skips[key]
+        if count > 0:
+            msg = msg if key != "Duplicate" and types_skipped < 2 else msg.replace(" &", ",")
+            msg += f"{prefixes[types_skipped]}{count} {key}"
+            types_skipped += 1
+
+    msg == "" or logger.warning(f"WARNING: {msg} Page(s) Skipped")
```

Comment:
I'd rely on the logger message configuration to indicate that this particular message is a warning, rather than including the string `WARNING: ` here. This allows the formatting to be handled in a centralized way and will ensure the messages are consistent.